src/assets/toolkit/scripts/lib/component/slideshow1.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCount = element.querySelector('.js-Slideshow-current'),
+    totalCount = element.querySelector('.js-Slideshow-total'),
+    classForward = 'slide-forward',
+    classForwardReset = 'slide-forwardReset',
+    classBack = 'slide-back'
+  } = {}) {
+
+    this.slides = Array.from(slides);
+    this.numSlides = this.slides.length;
+    this.counter = 0;
+    totalCount.innerHTML = this.numSlides;
+
+    Object.assign(this, {
+      element,
+      slideHolder,
+      nextTrigger,
+      prevTrigger,
+      currentCount,
+      totalCount,
+      classForward,
+      classForwardReset,
+      classBack
+    });
+
+    this.attachEvents();
+  }
+
+  updateCurrent () {
+    if (this.counter > 0) {
+      var currentSlide = (this.counter % this.numSlides);
+    } else {
+      var currentSlide = (this.counter % this.numSlides) ? ((this.counter % this.numSlides) + this.numSlides) : 0;
+    }
+
+    // Update the navigation with the current slide number
+    this.currentCount.innerHTML = currentSlide + 1;
+  }
+
+  getCurrentPosition (el) {
+    const transformMatrix = getComputedStyle(el).transform;
+    const slideHolderWidth = el.offsetWidth;
+    var matrixValues = transformMatrix.split('(')[1],
+    matrixValues = matrixValues.split(')')[0],
+    matrixValues = matrixValues.split(',');
+    const translateX = matrixValues[4];
+    const currentPosition = this.convertToPercent(translateX, slideHolderWidth);
+    return currentPosition;
+  }
+
+  convertToPercent(el, base) {
+    const percent = (el / base) * 100;
+    return percent;
+  }
+
+  findKeyframesRule(rule) {
+    var ss = document.styleSheets;
+    console.log(ss);
+    for (var j = 0; j < ss[1].cssRules.length; ++j) {
+      if (ss[1].cssRules[j].type == window.CSSRule.WEBKIT_KEYFRAMES_RULE &&
+      ss[1].cssRules[j].name == rule) {
+        return ss[1].cssRules[j]; }
+    }
+    return null;
```

Comment:
This is a route I was considering taking to change the keyframe properties dynamically. It got really complex and I abandoned it.